GitHubBranches.ps1

```diff
@@ -1072,128 +1074,1140 @@ filter Remove-GitHubRepositoryBranchProtectionRule
+        Retrieve a branch pattern protection rule for a given GitHub repository.
+
+    .DESCRIPTION
+        Retrieve a branch pattern protection rule for a given GitHub repository.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OwnerName
+        Owner of the repository.
+        If not supplied here, the DefaultOwnerName configuration property value will be used.
+
+    .PARAMETER RepositoryName
+        Name of the repository.
+        If not supplied here, the DefaultRepositoryName configuration property value will be used.
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER BranchPatternName
+        Name of the specific branch Pattern to be retrieved.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .INPUTS
+        GitHub.Branch
+        GitHub.Content
+        GitHub.Event
+        GitHub.Issue
+        GitHub.IssueComment
+        GitHub.Label
+        GitHub.Milestone
+        GitHub.PullRequest
+        GitHub.Project
+        GitHub.ProjectCard
+        GitHub.ProjectColumn
+        GitHub.Release
+        GitHub.Repository
+
+    .OUTPUTS
+        GitHub.BranchPatternProtectionRule
+
+    .EXAMPLE
+        Get-GitHubRepositoryBranchPatternProtectionRule -OwnerName microsoft -RepositoryName PowerShellForGitHub -BranchPatternName release/**/*
+
+        Retrieves branch protection rules for the release/**/* branch pattern of the PowerShellForGithub repository.
+
+    .EXAMPLE
+        Get-GitHubQlRepositoryBranchPatternProtectionRule -Uri 'https://github.com/microsoft/PowerShellForGitHub' -BranchPatternName master
+
+        Retrieves branch protection rules for the master branch pattern of the PowerShellForGithub repository.
+#>
+    [CmdletBinding(
+        PositionalBinding = $false,
+        DefaultParameterSetName = 'Elements')]
+    [OutputType({ $script:GitHubBranchProtectionRuleTypeName })]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSReviewUnusedParameter", "",
+        Justification="The Uri parameter is only referenced by Resolve-RepositoryElements which get access to it from the stack via Get-Variable -Scope 1.")]
+    param(
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            Position = 1,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'Uri')]
+        [Alias('RepositoryUrl')]
+        [string] $Uri,
+
+        [Parameter(
+            Mandatory,
+            Position = 2)]
+        [string] $BranchPatternName,
```

Comment:
It's not clear to me why this is required.  You're not using it in the query...it's part of post-processing.  It feels like it would be just as valuable to be able to query for all of the Branch Protection Rules on a repo as it would be to query for a single one.

Reply:
How about I make `BranchPatternName` non mandatory, and return all Branch Patterns if it is not specified?

Reply:
That's exactly what I was thinking.